### DISCUSSION TITLE
Capturing anything but `:` using @~ failed on `(`

### DISCUSSION THREAD

**Discussion by ethanyzhang** [alecthomas/participle#407] [Q&A] [answered]
Hello!

I have a ColumnHandle that I wanted to parse in this format:
```
<column name>:<data type>:<column index>
```
An example could be:
```
id:int:1
cohort:varchar(16):2
```
I defined my struct like:
```go
type HiveColumnHandle struct {
	ColumnName  string          `parser:"@Ident ':'"`
	DataType    string          `parser:"@~':' ':'"`
	ColumnIndex int             `parser:"@Int"`
}
```
I was hoping by specifying `@~':'`, I could parse `varchar(16)` into `DataType` but lexing stopped at `(` and complained that it is an unexpected token.

Could you help shed some light what is going wrong here?

Thanks a lot!!

**Comment by alecthomas** [comment, marked as answer]
Almost there, you're just missing a repetition: `parser:"@(~':')+ ':'"`
